### ISSUE TITLE
n_val in eval divides by number of batches

### ISSUE THREAD

**Issue by HansBambel** [milesial/Pytorch-UNet#171]
In your `eval_net` function you divide at the very end by `n_val` which is defined the following:
`n_val = len(loader)  # the number of batch`. Wouldn't it be better to divide by the number of samples instead, because we want to have the mean loss for every sample.

If, theoretically, we have all samples in a single batch we divide `tot` by 1, which would not be correct, right?

**Comment by milesial** [comment]
The default reduction for cross entropy for is `mean`. And for the Dice loss, it takes the average. So dividing by the batch size seems fine to me.

**Comment by HansBambel** [comment]
True, you're right.
